### DISCUSSION TITLE
Create Record without Authentication

### DISCUSSION THREAD

**Discussion by thisisfel1x** [pocketbase/pocketbase#3998] [Q&A] [answered]
Hey everyone,

I have a question regarding the creation of records (using JS SDK).
My plan is that a user can submit a booking form (with data like name, email, date, number of people etc) without an account. This data should then be saved in a record.

An admin or similar can then process, delete and manage the requests. He is logged into his account

How can I implement this system, i.e. the creation of records, without authorization? My main concern is security, so that an infinite number of requests are not created by an exploit or other means. 

I would be grateful for any tips. The question may seem trivial, but my experience in this area is still limited.

Best regards

**Comment by jonshipman** [comment, marked as answer]
Under the API rules just unlock the "Create" function. If you don't enter any restraints, the table can be added by to anyone.

Some different processes can be used to limit SPAM. Personally, I like avoiding exposing my API routes. Maybe it's because I have more familiarity with the node environment as opposed to go, but I will wrap my pocketbase requests to come from the server hosting the frontend. I will usually lock pocketbase's admin UI behind my ip address. This is all just personal preference. Here's my boilerplate for svelte projects https://github.com/jonshipman/sveltekit-pb-boilerplate

You will get SPAM. The best method for dealing with it would be an email magic link. Create a record, `approved` boolean field set to false, email the person inputting the data, click the link, `approved` = true. Similar to the signup route but avoiding the user management. Just have a `cronAdd` function in pocketbase that deletes unapproved data older than 1hour. 

The next method would be a word blacklist. I have a client running a 10 year old system I wrote and he's been using a word blacklist all that time. Requires a bit more management on your part (or another admin) but he seems happy with it.

Google recaptcha will stop bots but not people paid to fill out forms, so YMMV. It stops most, but some of the biggest clients I manage still get 5 or so spam messages a day with Recaptcha. 

All in all, most websites I setup will use Recaptcha. It's the easiest to manage and most don't see Spam unless you get millions of visitors a month. A blacklist will work if someone puts in elbow grease to manage it (you should capture the ones that fail just to see if any false positives are caught, but dump them using cronAdd). The best method would be the email magic link. While, yes, it's easy to setup a free email account I personally haven't seen any come across my desk in the 16 years or so I've been working. Unless you're working for a big site like Walmart or Disney you should be fine.
